src/components/Dashboard/Dashboard.story.jsx

```diff
@@ -0,0 +1,1076 @@
+          ? { interval: 'day', period: 'month' }
+          : range === 'thisQuarter'
+          ? { interval: 'week', period: 'quarter' }
+          : range === 'thisYear'
+          ? { interval: 'month', period: 'year' }
+          : { interval: 'day', num: 7 };
+
+      setCards(
+        cards.map(i =>
+          i.id === id
+            ? {
+                ...i,
+                interval: cardRange.interval,
+                timeRange: range,
+                values: cardRange.period
+                  ? getPeriodChartData(
+                      cardRange.interval,
+                      cardRange.period,
+                      { min: 10, max: 100 },
+                      100
+                    )
+                  : getIntervalChartData(
+                      cardRange.interval,
+                      cardRange.num,
+                      { min: 10, max: 100 },
+                      100
+                    ),
+              }
+            : i
+        )
+      );
+    }
+    if (type === 'table_card_row_action') {
+      console.log(id, type, payload);
+    }
+  };
+
+  /*
+  return (
+    <div>
+      <Button style={{ margin: '20px 0 0 20px' }} onClick={handleAdd}>
+        Add card
+      </Button>
+      <Dashboard cards={cards} onCardAction={handleCardAction} {...props} />
+    </div>
+  );
+  */
+  return <Dashboard cards={cards} onCardAction={handleCardAction} {...props} />;
+};
+
+storiesOf('Watson IoT Experimental|Dashboard', module)
+  .add('basic', () => {
+    return (
+      <FullWidthWrapper>
+        <StatefulDashboard
+          title={text('title', 'Munich Building')}
+          lastUpdated={Date()}
+          isEditable={boolean('isEditable', false)}
+          isLoading={boolean('isLoading', false)}
+          onBreakpointChange={action('onBreakpointChange')}
+          onLayoutChange={action('onLayoutChange')}
+        />
+      </FullWidthWrapper>
+    );
+  })
+  .add('basic - without last updated header', () => {
+    return (
+      <FullWidthWrapper>
+        <StatefulDashboard
+          title={text('title', 'Munich Building')}
+          isEditable={boolean('isEditable', false)}
+          isLoading={boolean('isLoading', false)}
+          onBreakpointChange={action('onBreakpointChange')}
+          onLayoutChange={action('onLayoutChange')}
+          hasLastUpdated={false}
+        />
+      </FullWidthWrapper>
+    );
+  })
+  .add('custom actions', () => {
+    return (
+      <FullWidthWrapper>
+        <StatefulDashboard
+          title={text('title', 'Munich Building')}
+          isEditable={boolean('isEditable', false)}
+          isLoading={boolean('isLoading', false)}
+          onBreakpointChange={action('onBreakpointChange')}
+          onLayoutChange={action('onLayoutChange')}
+          actions={[{ id: 'edit', label: 'Edit', icon: 'edit' }]}
+          onDashboardAction={action('onDashboardAction')}
+          hasLastUpdated={false}
+        />
+      </FullWidthWrapper>
+    );
+  })
```

Comment:
This story is broken in the preview https://deploy-preview-579--carbon-addons-iot-react.netlify.com/?path=/story/watson-iot-experimental-dashboard--custom-actions

```
TypeError: Cannot read property 'height' of undefined
    at Icon
```

Reply:
fixed.